simpleclient_logback/src/main/java/io/prometheus/client/logback/InstrumentedAppender.java

```diff
@@ -0,0 +1,84 @@
+package io.prometheus.client.logback;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.core.UnsynchronizedAppenderBase;
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.Counter;
+
+public class InstrumentedAppender extends UnsynchronizedAppenderBase<ILoggingEvent> {
+
+  public static final String COUNTER_NAME = "logback_appender_total";
+
+  private final CollectorRegistry registry;
+
+  private Counter counter;
+  private Counter.Child trace;
+  private Counter.Child debug;
+  private Counter.Child info;
+  private Counter.Child warn;
+  private Counter.Child error;
+
+  /**
+   * Create a new instrumented appender using the default registry.
+   */
+  public InstrumentedAppender() {
+    this(CollectorRegistry.defaultRegistry);
+  }
+
+  /**
+   * Create a new instrumented appender using the given registry.
+   *
+   * @param registry the collector registry
+   */
+  public InstrumentedAppender(CollectorRegistry registry) {
+    this.registry = registry;
+    setName(Appender.class.toString());
+  }
+
+  @Override
+  public void start() {
```

Comment:
Even across unittests that reinitilize the entire logging setup?

If there's any cases where this can be called twice or this class instantiated twice then it'll break scraping the entire target, so I think we should play this one safe.